lib/src/tools/format_tool.dart

```diff
@@ -178,135 +191,51 @@ class FormatTool extends DevTool {
 
     final dir = Directory(root ?? '.');
 
-    // Use Glob.listSync to get all directories which might include a matching file.
-    var directoriesWithExcludes = <String>{};
-
-    if (collapseDirectories) {
-      for (var g in exclude) {
-        List<FileSystemEntity>? matchingPaths;
-        try {
-          matchingPaths = g.listSync(followLinks: followLinks);
-        } on FileSystemException catch (_) {
-          _log.finer("Glob '$g' did not match any paths.\n");
-        }
-        if (matchingPaths != null) {
-          for (var path in matchingPaths) {
-            if (path is Directory) {
-              directoriesWithExcludes.add(path.path);
-            } else {
-              directoriesWithExcludes.add(path.parent.path);
-            }
-          }
-        }
-      }
-
-      // This is all the directories that contain a match within them.
-      _log.finer("Directories with excludes:\n");
-      for (var dir in directoriesWithExcludes) {
-        _log.finer("  $dir\n");
-      }
-      _log.finer(
-          "${directoriesWithExcludes.length} directories contain excludes\n");
-    }
-
-    String currentDirectory = p.relative(dir.path, from: dir.path);
-    bool skipFilesInDirectory = false;
-    for (final entry
-        in dir.listSync(recursive: true, followLinks: followLinks)) {
-      final relative = p.relative(entry.path, from: dir.path);
-      _log.finest('== Processing relative $relative ==\n');
-
-      if (p.isWithin(currentDirectory, relative)) {
-        if (skipFilesInDirectory) {
-          _log.finest('skipping child $entry\n');
-          continue;
-        }
-      } else {
-        // the file/dir in not inside, cancel skipping.
-        skipFilesInDirectory = false;
-      }
+    for (final entry in _listSyncWithoutHidden(dir,
+        recursive: true, followLinks: followLinks)) {
+      final filename = p.relative(entry.path, from: dir.path);
+      _log.finest('== Processing relative $filename ==\n');
 
       if (entry is Link) {
-        _log.finer('skipping link $relative\n');
-        skippedLinks.add(relative);
+        _log.finer('skipping link $filename\n');
```

Comment:
Why did these checks not make it into the listSync forEach?  Would they even be faster there?

Reply:
That would definitely be faster— I chose to keep this closer to the original logic for now to avoid churn. I tentatively plan to make another pass to reduce the amount of work being done in this function later.